pytorch_lightning/utilities/fetching.py

```diff
@@ -57,48 +55,30 @@ def fetching_function(self) -> Any:
         """Override with your own fetching logic."""
 
     @abstractmethod
-    def prefetching(self, prefetch_batches: int) -> None:
+    def prefetching(self) -> None:
         """Override with your own pre-fetching logic."""
 
-    def __init__(
-        self,
-        prefetch_batches: int = 0,
-    ) -> None:
+    def __init__(self, prefetch_batches: int = 0) -> None:
         if prefetch_batches < 0:
             raise MisconfigurationException("`prefetch_batches` should at least be 0.")
-
-        self.store_on_device = False
-        self.prefetch_batches = prefetch_batches + 1
+        self.prefetch_batches = prefetch_batches
 
         self.dataloader: Optional[Union[DataLoader, CombinedLoader]] = None
         self.dataloader_iter: Optional[Iterator] = None
 
-        self.stage: Optional[str]
         self.batch_to_device: Optional[Callable]
-        self.profiler: "Optional[pl.profiler.base.BaseProfiler]"
 
         self.batches: List
         self.fetched: int
         self.done: bool
 
         self.reset()
 
-    def setup(
-        self,
-        dataloader: Iterable,
-        stage: Optional[str] = None,
-        batch_to_device: Optional[Callable] = None,
-        profiler: "Optional[pl.profiler.base.BaseProfiler]" = None,
-    ) -> None:
+    def setup(self, dataloader: Iterable, batch_to_device: Optional[Callable] = None) -> None:
         self._add_capture_metadata_collate(dataloader)
 
         self.dataloader = dataloader
-        self.stage = stage
         self.batch_to_device = batch_to_device
-        self.profiler = profiler
-
-        if self.profiler is not None and stage is None:
```

Comment:
Because `call_hook` already integrates the profiler and I don't think this needs to also profile each `next`. If this is requested by users, we can think about what's the cleanest way to add it back.